asf_search/ASFProduct.py

```diff
@@ -44,7 +45,17 @@ def download(self, path: str, filename: str = None, session: ASFSession = None,
         """
         if filename is None:
             default_filename = self.properties['fileName']
-        
+        elif filename is not None:
+            multiple_files = (
+                (fileType == FileDownloadType.ADDITIONAL_FILES and len(self.properties['additionalUrls']) > 1) 
+                or fileType == FileDownloadType.ALL_FILES
+            )
+            if multiple_files:
+                ASF_LOGGER.warning(f"Attempting to download multiple files for product, ignoring user provided filename argument \"{filename}\", using default.")
+                default_filename = self.properties['fileName']
```

Comment:
And if you make the default_filename change above, you won't have to re-set it here